miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppCustomPermissionCache.kt

```diff
@@ -0,0 +1,45 @@
+package com.rakuten.tech.mobile.miniapp.js
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.rakuten.tech.mobile.miniapp.js.MiniAppCustomPermissionResult.CUSTOM_PERMISSION_DENIED
+
+/**
+ * A class to check if a requested permission has been granted or rejected with storing
+ * the values of the grant result per permission.
+ */
+internal class MiniAppCustomPermissionCache(context: Context) {
+    @VisibleForTesting
+    val prefs: SharedPreferences = context.getSharedPreferences(
+        "com.rakuten.tech.mobile.miniapp.permissions", Context.MODE_PRIVATE
+    )
+
+    /**
+     * Reads grant result from SharedPreferences.
+     * @param [permissions] the key-set provided to find the stored corresponding grant results.
+     * @return [List<String>] the grant results of permissions requested.
+     */
+    fun readPermissions(permissions: List<String>): List<String> {
+        val grantResults = mutableListOf<String>()
+
+        permissions.forEach { key ->
+            prefs.getString(key, CUSTOM_PERMISSION_DENIED)?.let { grantResults.add(it) }
+        }
+
+        return grantResults
```

Comment:
Rather than returning a list containing only the permission result, maybe it's more useful to return a list of all requested permissions along with the `MiniAppCustomPermissionResult` status?

So instead this function could return `List<Pair< MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>`. Or maybe even a custom type with `permission` and `result` keys? It seems like this would be more convenient for the users. What do you think?